pkg/typesystem/typesystem.go

```diff
@@ -862,6 +869,10 @@ func NewAndValidate(ctx context.Context, model *openfgav1.AuthorizationModel) (*
 		return nil, err
 	}
 
+	if err := t.validateConditions(); err != nil {
```

Comment:
Should we add some tests to `typesystem_test.go` to ensure this new behavior continues to behave as expected, regardless of what happens upstream in WriteAuthorizationModel RPC?

Since many code paths use `typesystem.NewAndValidate` I'd like to make sure we have the coverage directly alongside the code instead of indirectly through some other related code path.